docs/Lexicon.rst

```diff
@@ -235,6 +269,10 @@ source code, tests, and commit messages. See also the `LLVM lexicon`_.
     IR generation.
     See `mandatory passes <mandatory passes / mandatory optimizations>`.
 
+  reabstraction
+    An implicit representation change that occurs when a value is used with
+    a different `abstraction pattern` from its current representation.
```

Comment:
Interesting question. We do use the "abstraction pattern" concept to govern method representations, so variance-related thunks could indeed be considered reabstractions.